sdks/python/apache_beam/transforms/window.py

```diff
@@ -392,9 +392,11 @@ def __init__(self, size, period, offset=0):
 
   def assign(self, context):
     timestamp = context.timestamp
-    start = timestamp - (timestamp - self.offset) % self.period
-    return [IntervalWindow(Timestamp.of(s), Timestamp.of(s) + self.size)
-            for s in range(start, start - self.size, -self.period)]
+    start = timestamp - ((timestamp - self.offset) % self.period)
+    return [
+        IntervalWindow(Timestamp(micros=s), Timestamp(micros=s) + self.size)
```

Comment:
I wouldn't change the return type, just remove it altogether. It just means one couldn't write int(timestamp) and get the (implicitly truncated) timestamps anymore, including using them in range. 

I don't think anyone is using it (and if they are, likely there's a bug like above).